src/explore-education-statistics-common/src/modules/table-tool/components/__tests__/DataSetStep.test.tsx

```diff
@@ -0,0 +1,582 @@
+    expect(radios[0]).toEqual(
+      screen.getByRole('radio', { name: 'View all featured tables' }),
+    );
+    expect(radios[1]).toEqual(screen.getByRole('radio', { name: 'Subject 1' }));
+    expect(radios[2]).toEqual(screen.getByRole('radio', { name: 'Subject 2' }));
+
+    expect(
+      screen.getByText(
+        'Please select a data set, you will then be able to see a summary of the data, create your own tables, view featured tables, or download the entire data file.',
+      ),
+    ).toBeInTheDocument();
+  });
+
+  test('shows the correct step heading and the `view all featured tables` option if the release has featured tables', () => {
+    render(
+      <DataSetStep
+        {...wizardProps}
+        featuredTables={testFeaturedTables}
+        renderFeaturedTable={highlight => <a href="/">{highlight.name}</a>}
+        subjects={testSubjects}
+        onSubmit={noop}
+      />,
+    );
+
+    expect(
+      screen.getByRole('heading', {
+        name: 'Step 1 (current) Select a data set or featured table',
+      }),
+    ).toBeInTheDocument();
+
+    const radios = within(
+      screen.getByRole('group', {
+        name: 'View all featured tables or select a data set',
+      }),
+    ).getAllByRole('radio');
+    expect(radios).toHaveLength(3);
+    expect(radios[0]).toEqual(
+      screen.getByRole('radio', { name: 'View all featured tables' }),
+    );
+    expect(radios[1]).toEqual(screen.getByRole('radio', { name: 'Subject 1' }));
+    expect(radios[2]).toEqual(screen.getByRole('radio', { name: 'Subject 2' }));
+  });
+
+  test('shows the correct step heading and does not show the `view all featured tables` option if the release has no featured tables', () => {
+    render(
+      <DataSetStep
+        {...wizardProps}
+        renderFeaturedTable={highlight => <a href="/">{highlight.name}</a>}
+        subjects={testSubjects}
+        onSubmit={noop}
+      />,
+    );
+
+    expect(
+      screen.getByRole('heading', {
+        name: 'Step 1 (current) Select a data set',
+      }),
+    ).toBeInTheDocument();
+
+    expect(
+      screen.queryByLabelText('View all featured tables'),
+    ).not.toBeInTheDocument();
+
+    const radios = within(
+      screen.getByRole('group', {
+        name: 'Select a data set',
+      }),
+    ).getAllByRole('radio');
+    expect(radios).toHaveLength(2);
+    expect(radios[0]).toEqual(screen.getByRole('radio', { name: 'Subject 1' }));
+    expect(radios[1]).toEqual(screen.getByRole('radio', { name: 'Subject 2' }));
+  });
+
+  test('clicking `view all featured tables` shows all featured tables for the release', () => {
+    render(
+      <DataSetStep
+        {...wizardProps}
+        featuredTables={testFeaturedTables}
+        renderFeaturedTable={highlight => <a href="/">{highlight.name}</a>}
+        subjects={testSubjects}
+        onSubmit={noop}
+      />,
+    );
+
+    userEvent.click(
+      screen.getByRole('radio', { name: 'View all featured tables' }),
+    );
+
```

Comment:
Might be worth a simple check that the text from the previous view is no longer visible